packages/features/ee/organizations/pages/settings/getServerSidePropsMembers.tsx

```diff
@@ -0,0 +1,36 @@
+import type { GetServerSidePropsContext } from "next";
+
+import { getServerSession } from "@calcom/features/auth/lib/getServerSession";
+import prisma from "@calcom/prisma";
+import type { inferSSRProps } from "@calcom/types/inferSSRProps";
+
+export type PageProps = inferSSRProps<typeof getServerSidePropsForMembers>;
+
+export const getServerSidePropsForMembers = async function getServerSidePropsForMembers({
+  req,
+  res,
+}: GetServerSidePropsContext) {
+  const session = await getServerSession({ req, res });
+
+  if (!session || !session.user) {
+    return {
+      redirect: {
+        permanent: false,
+        destination: "/auth/login",
+      },
+    };
+  }
+
+  if (!session.user.org?.id) {
```

Comment:
https://github.com/calcom/cal.com/blob/8a972a74c67bb11d3833a733e054aa0e182cddcf/packages/features/auth/lib/getServerSession.ts#L38-L40

When we decode token here, token.org is undefined for users with organization, but only when doing it via e2e, works fine normally.

Couldn't really figure out why till now, it is kinda confusing. As a work around we can use profile.organizationId instead of org?.id which I checked is working for both e2e and webapp. @hariombalhara . Will keep digging till then 🤔

```suggestion
  if (!session.user.profile?.organizationId) {
```

Reply:
Update:

https://github.com/calcom/cal.com/blob/d84f08c998623f0fbb5975a684a8830327294260/packages/features/auth/lib/next-auth-options.ts#L229-L238

My best guess is when doing .apiLogin, CredentialsProvider block is what runs where we are never adding org. 

https://github.com/calcom/cal.com/blob/d84f08c998623f0fbb5975a684a8830327294260/packages/features/auth/lib/next-auth-options.ts#L555-L574 . So user.org is always undefined when running e2e

Reply:
Ig if we want to fix the core issue, we can call `autoMergeIdentites`

https://github.com/calcom/cal.com/blob/d84f08c998623f0fbb5975a684a8830327294260/packages/features/auth/lib/next-auth-options.ts#L469
 

for account.type === "credentials" as well


https://github.com/calcom/cal.com/blob/d84f08c998623f0fbb5975a684a8830327294260/packages/features/auth/lib/next-auth-options.ts#L560-L574

Honestly I am not completely sure what any of these do so I will leave it up to you @hariombalhara

Reply:
If it's a testcase thing only it is better to just use what works for now. I am good with the changes.